ndonnx/_core/_numericimpl.py

```diff
@@ -16,99 +18,88 @@
 import ndonnx._opset_extensions as opx
 from ndonnx._utility import promote
 
-from ._interface import OperationsBlock
+from ._default import OperationsBlock
+from ._utils import (
+    binary_op,
+    from_corearray,
+    split_nulls_and_values,
+    unary_op,
+    variadic_op,
+    via_upcast,
+)
 
 if TYPE_CHECKING:
-    from ndonnx._array import Array
+    from ndonnx import Array
     from ndonnx._corearray import _CoreArray
 
 
-class CoreOperationsImpl(OperationsBlock):
+class NumericOperationsImpl(OperationsBlock):
     # elementwise.py
 
     def abs(self, x):
-        return _unary_op(x, opx.abs)
+        return unary_op(x, opx.abs)
```

Comment:
I realize there is still the eager propagation behind this function, but is there anything conceptually wrong with doing:
```python
        import spox.opset.ai.onnx.v21 as op
        return ndx.from_spox_var(op.abs(x.spox_var()))
```

This would feel correct to me as a default. If we want to do any dtype specific logic, it should be implemented in dtype-specific subclass of `NumericOperationsImpl`, shouldn't it? Meanwhile, I wonder if we can't delegate the eager propagation up the stack (closer) to the public interface.

Reply:
> Meanwhile, I wonder if we can't delegate the eager propagation up the stack (closer) to the public interface.

I think we most certainly can move eager propagation up the stack (to the exposed function level I imagine like `ndx.add`). You would sacrifice the granularity at which your debugger can help you since stepping into the function wouldn't be super helpful, but it might be a valid trade off of discuss.

> If we want to do any dtype specific logic, it should be implemented in dtype-specific subclass of NumericOperationsImpl, shouldn't it?

Yes but can we revisit this in a subsequent PR? I think this is a separate item of work and wouldn't affect the public API.